core/src/main/java/org/apache/spark/util/collection/unsafe/sort/PrefixComparators.java

```diff
@@ -28,12 +28,31 @@
   private PrefixComparators() {}
 
   public static final PrefixComparator STRING = new UnsignedPrefixComparator();
+  public static final PrefixComparator STRING_NULLLAST = new UnsignedPrefixComparatorNullLast();
+
+  public static final PrefixComparator STRING_DESC_NULLFIRST =
+          new UnsignedPrefixComparatorDescNullFirst();
   public static final PrefixComparator STRING_DESC = new UnsignedPrefixComparatorDesc();
+
   public static final PrefixComparator BINARY = new UnsignedPrefixComparator();
+  public static final PrefixComparator BINARY_NULLLAST = new UnsignedPrefixComparatorNullLast();
+
+  public static final PrefixComparator BINARY_DESC_NULLFIRST =
+          new UnsignedPrefixComparatorDescNullFirst();
   public static final PrefixComparator BINARY_DESC = new UnsignedPrefixComparatorDesc();
+
   public static final PrefixComparator LONG = new SignedPrefixComparator();
+  public static final PrefixComparator LONG_NULLLAST = new SignedPrefixComparatorNullLast();
+
+  public static final PrefixComparator LONG_DESC_NULLFIRST =
+          new SignedPrefixComparatorDescNullFirst();
   public static final PrefixComparator LONG_DESC = new SignedPrefixComparatorDesc();
```

Comment:
Probably better to order as LONG, LONG_DESC, LONG_NULL_LAST, LONG_DESC_NULL_LAST